src/Data/List/Base.agda

```diff
@@ -187,6 +187,10 @@ replicate : ℕ → A → List A
 replicate zero    x = []
 replicate (suc n) x = x ∷ replicate n x
 
+iterate : (A → A) → A → ℕ → List A
+iterate f k zero    = []
```

Comment:
I wouldn't give the argument of type `A` the name `k`... I think that stdlib convention has that as either `x` (pros and cons either way), or else `a` (but often reserved as the level of `A`... so perhaps, sadly, best avoided). 

To me, `k` is always a `Nat`, maybe a `Fin`, or even an `Integer`...

Reply:
Or, for that matter, take the `s` and `z` names from `Vec.iterate`? These are, after all, operations derived from the canonical NNO structure on`Nat`...

Reply:
Thanks for the review! I've updated the argument names to match `Vec.Base.iterate`.